taskcluster/docker/node/build.py

```diff
@@ -73,17 +73,34 @@ def write_package_info(package_info):
 
 def get_buildid():
     now = datetime.utcnow()
-    return now.strftime("%Y%m%d.%H%M%S")
+    # note the `-` (hyphen) in `%-H`
+    # this removes the leading zero
+    # from the hour (leading zeros are not allowed)
+    return now.strftime("%Y%m%d.%-H%M%S")
 
 
 def get_buildid_version(version):
-    """Version schema check+append a `buildid{buildid}` to ensure unique version."""
-    # XXX is there a more precise schema we should verify ourselves against?
-    if len(version.split(".")) not in (1, 2, 3):
-        raise Exception("{version} has too many version parts!")
-    if "buildid" in version:
-        raise Exception("{version} already has a buildid specified!")
-    buildid_version = f"{version}buildid{get_buildid()}"
+    """Check the version's formating and append `date.time` as a buildid to ensure a unique version.
+    The addon's in-tree extension/manifest.json file specifies `major.minor`.
```

Comment:
note: it's actually `package.json`

Reply:
is this documented anywhere?  Maybe we could add a new section about extension versions in https://github.com/mozilla-extensions/xpi-manifest/blob/main/docs/adding-a-new-xpi.md#creating-the-official-repo?

Reply:
This is actually confusing. Is the version Firefox cares about the one in the manifest? This bit of documentation from [the migration guide](https://extensionworkshop.com/documentation/develop/manifest-v3-migration-guide/) seems to say so.

> **Extension version in the manifest**
>
> The format of the top-level manifest.json version key in Firefox has evolved and became simpler: letters and other previously allowed symbols are no longer accepted. The value must be a string with 1 to 4 numbers separated by dots (e.g., 1.2.3.4). Each number can have up to 9 digits and leading zeros before another digit are not allowed (e.g., 2.01 is forbidden, but 0.2, 2.0.1, and 2.1 are allowed)

Reply:
Never mind. I see what is going on. The code reads the version from package dot json appends a build id - then updates the version in both package dot json and manifest dot json to the new build id version.